apps/extension/src/provider/InjectedAnoma.ts

```diff
@@ -0,0 +1,177 @@
+import { ChainInfo as Chain } from "@keplr-wallet/types";
+import { Anoma as IAnoma, Signer } from "@anoma/types";
+import { Anoma } from "./Anoma";
+import { Result } from "../router/types";
+
+type ProxyMethods =
+  | "suggestChain"
+  | "connect"
+  | "getSigner"
+  | "chain"
+  | "chains"
+  | "version";
```

Comment:
nit:
```suggestion
enum ProxyMethod {
  SuggestChain = "suggestChain",
  Connect = "connect"
}
```
I feel that while needing a few more keystrokes at use site, this makes a way better refactoring experience as one can much more confidently do a global search 

`ProxyMethod.Connect` vs. `connect`

Reply:
I'll take a look. These values map 1-to-1 with the naming used in the `Anoma` and `InjectedAnoma` methods, so I wonder if there's a better mechanism to confirm the passed method arg matches this (though I'm not opposed to wrapping them in an enum, but it would be nice to not have to worry about it while refactoring at all if possible if they could be typed more automatically). If I don't find a better way, I'll move them into an enum

Reply:
As the methods we want to proxy belong to the `Anoma` class, I've simply done this:
```ts
type ProxyMethod = keyof Anoma;
```
Which gives us the type-safety we need, without need for future maintenance.

Reply:
Even better, I've added:
```ts
type ProxyMethod = keyof Omit<Anoma, "version">;
```
As we only want to type-check methods that can actually get proxied. During runtime, this has no effect, but makes it clear to developers _which_ methods should be allowed to pass through.